### DISCUSSION TITLE
Does mediamtx have any issues related to H.264 licensing?

### DISCUSSION THREAD

**Discussion by Allen48953** [bluenviron/mediamtx#3985] [Questions]
### Question

Hello everyone,

Thank you for your support, which enables us to enjoy such convenient multimedia services. I have a couple of questions: Are there any H.264 licensing issues for a server running mediamtx when the stream codec is H.264? Additionally, does mediamtx handle encoding or decoding of H.264 stream? Your insights on these matters would be greatly appreciated.

Thank you.

**Comment by aler9** [comment]
Hello, we do not decode or encode streams, we just route them from the source to any connected readers. Therefore, no patented technology is used and no license is required. Licensing is required when encoding or decoding H264. If you have an hardware encoder/decoder, like a GPU or CPU, license costs are included in hardware cost.
